### PR TITLE
Restrict gssapi flag to only host platform

### LINKED ISSUE #13322: make build-cross fails on ppc64le platform
[provide a description of the issue]

##### Version
openshift master branch

##### Steps To Reproduce
1. Clone openshift origin master branch
2. make build-cross

##### Current Result
```
[root@localhost origin]# make build-cross
rm -rf _output
hack/build-cross.sh
++ Building go targets for linux/ppc64le: images/pod cmd/dockerregistry cmd/gitserver pkg/sdn/plugin/sdn-cni-plugin vendor/github.com/containernetworking/cni/plugins/ipam/host-local vendor/github.com/containernetworking/cni/plugins/main/loopback examples/hello-openshift examples/deployment
++ Building go targets for linux/amd64: cmd/openshift cmd/oc
# github.com/openshift/origin/pkg/cmd/util/tokencmd
pkg/cmd/util/tokencmd/negotiator_gssapi.go:25: undefined: gssapi.Lib
pkg/cmd/util/tokencmd/negotiator_gssapi.go:32: undefined: gssapi.Name
pkg/cmd/util/tokencmd/negotiator_gssapi.go:34: undefined: gssapi.CtxId
pkg/cmd/util/tokencmd/negotiator_gssapi.go:43: undefined: gssapi.CredId
[ERROR] PID 25650: hack/common.sh:153: `local -a binaries=("$@")` exited with status 1.
[INFO] 		Stack Trace: 
[INFO] 		  1: hack/common.sh:153: `local -a binaries=("$@")`
[INFO] 		  2: hack/build-cross.sh:81: os::build::build_binaries
[INFO]   Exiting with code 1.
make: *** [build-cross] Error 1
[root@localhost origin]#
```
##### Expected Result

make build-cross should pass

##### Additional Information
[try to run `$ oadm diagnostics` or `oc adm diagnostics` command if possible]
[if you are reporting issue related to builds, provide build logs with `BUILD_LOGLEVEL=5`]
[consider attaching output of the `$ oc get all -o json -n <namespace>` command to the issue]
[visit https://docs.openshift.org/latest/welcome/index.html]

### PR DESCRIPTION
**Pull request by mkumatag**
Fixes #13322

### CODE REVIEW

File: hack/build-cross.sh [hunk 1]
```diff
@@ -8,8 +8,7 @@ host_platform="$(os::build::host_platform)"
 
 # Set build tags for these binaries
 readonly OS_GOFLAGS_TAGS="include_gcs include_oss"
-readonly OS_GOFLAGS_TAGS_LINUX_PPC64LE="gssapi"
-readonly OS_GOFLAGS_TAGS_LINUX_AMD64="gssapi"
+readonly OS_GOFLAGS_TAGS_$(echo ${host_platform} | tr '[:lower:]/' '[:upper:]_')="gssapi"
```
Comment (stevekuznetsov) [inline hack/build-cross.sh]: There are a number of places (also in `common.sh`) that use this logic to get from platform --> tag envar name. Can you factor that into a function so the logic is a little more unified?

Comment (mkumatag) [inline hack/build-cross.sh]: @stevekuznetsov addressed above comment, PTAL into latest commit.

**Comment by stevekuznetsov** [comment]
@tdawson you might want to review this

**Comment by tdawson** [comment]
Manually tested this for an aarch64 build.  Works great.  Thanks for letting me take a look at it so I could have some peace of mind.

**Comment by stevekuznetsov** [comment]
[merge]

**Pull request merged**